apps/blockchain/lib/blockchain/block.ex

```diff
@@ -447,12 +447,14 @@ defmodule Blockchain.Block do
     beneficiary = opts[:beneficiary] || nil
     extra_data = opts[:extra_data] || <<>>
     state_root = opts[:state_root] || parent_block.header.state_root
+    mix_hash = opts[:mix_hash] || parent_block.header.mix_hash
```

Comment:
Interesting catch. I've been trying to track a mismatching mix_hash for another test. I wonder if this will help!